tools/upgrade.sh

```diff
@@ -34,6 +31,41 @@ then
   printf "${BLUE}%s\n" "Hooray! Oh My Zsh has been updated and/or is at the current version."
   printf "${BLUE}${BOLD}%s${NORMAL}\n" "To keep up on the latest news and updates, follow us on twitter: https://twitter.com/ohmyzsh"
   printf "${BLUE}${BOLD}%s${NORMAL}\n" "Get your Oh My Zsh swag at:  http://shop.planetargon.com/"
-else
+}
+
+error_upgrading() {
   printf "${RED}%s${NORMAL}\n" 'There was an error updating. Try again later?'
+  exit 1
+}
+
+run_upgrade() {
+  if git pull --rebase --stat origin master
+  then
+    success_upgrading
+  else
+    error_upgrading
+  fi
+}
+
+printf "${BLUE}%s${NORMAL}\n" "Updating Oh My Zsh"
+cd "$ZSH"
+
+if output=$(git status --porcelain); then
```

Comment:
Is this the right test? `git status --porcelain` has an exit status of zero even if there are changes, which is what this `if output=$(...); then` is testing. Maybe you want to capture its output and test if that is a nonempty string?